File: src/main/java/com/elf/vipForElf/domain/test/repository/JPA/FloorJPARepository.java
package com.elf.vipForElf.domain.test.repository.JPA;

import com.elf.vipForElf.domain.test.entity.FloorEntity;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface FloorJPARepository extends JpaRepository<FloorEntity,Long> {

    Page<FloorEntity> findByBuildingIdOrFloorNumberOrPurposeIgnoreCase(Long buildingId, Integer floorNumber, String purpose, Pageable pageable);
}

File: src/main/java/com/elf/vipForElf/domain/floor/repository/JPA/FloorJPARepository.kt
package com.elf.vipForElf.domain.floor.repository.JPA

import com.elf.vipForElf.domain.floor.entity.FloorEntity
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface FloorJPARepository : JpaRepository<FloorEntity, Long> {
    fun findByBuildingIdOrFloorNumberOrPurposeIgnoreCase(
        buildingId: Long?,
        floorNumber: Int?,
        purpose: String?,
        pageable: Pageable
    ): Page<FloorEntity>
}
